manifests/vhost/custom.pp

```diff
@@ -0,0 +1,38 @@
+# See README.md for usage information
+define apache::vhost::custom(
+  $content,
+  $ensure = 'present',
+  $priority = '25',
+) {
+  include ::apache
+
+  ## Apache include does not always work with spaces in the filename
+  $filename = regsubst($name, ' ', '_', 'G')
+
+  ::apache::custom_config { $filename:
+    ensure   => $ensure,
+    confdir  => $::apache::vhost_dir,
+    content  => $content,
+    priority => $priority,
+  }
+
+  # NOTE(pabelanger): This code is duplicated in ::apache::vhost and needs to
+  # converted into something generic.
+  if $::apache::vhost_enable_dir {
+    $vhost_symlink_ensure = $ensure ? {
+      present => link,
+      default => $ensure,
+    }
+
+    file { "${priority}-${filename}.conf symlink":
+      ensure  => $vhost_symlink_ensure,
+      path    => "${::apache::vhost_enable_dir}/${priority}-${filename}.conf",
+      target  => "${::apache::vhost_dir}/${priority}-${filename}.conf",
+      owner   => 'root',
+      group   => $::apache::params::root_group,
```

Comment:
This doesn't look like it supports the same root group override that the base apache class does.  Is that the plan?  Also what about the user for managing it?  Would it always be 'root' even if overridden in ::apache?